Cabal/Distribution/Package.hs

```diff
@@ -175,7 +175,7 @@ instance NFData Module where
 -- @since 2.0
 
 data ComponentId
-    = ComponentId String
+    = ComponentId ShortText
```

Comment:
@hvr I believe component IDs are actually ASCII only (indeed a small subset of ASCII), so you could use ShortByteString.

@ezyang can you confirm and tell us what the characters are so we can document that here (and perhaps add a validation function / assertion).